python/ray/serve/_private/controller.py

```diff
@@ -929,10 +929,15 @@ def get_serve_instance_details(self) -> Dict:
         # Eventually we want to remove route_prefix from DeploymentSchema.
         http_options = HTTPOptionsSchema.parse_obj(http_config.dict(exclude_unset=True))
         grpc_options = gRPCOptionsSchema.parse_obj(grpc_config.dict(exclude_unset=True))
+        proxy_location = (
```

Comment:
Non-blocker: might be good to add a comment to explain the differences between `DeploymentMode` and `ProxyLocation` so future folks know why this is needed and how it fixes the issue.

Reply:
I added a docstring to `_from_deployment_mode`.